### PR TITLE
make compatible with puppet 4.2.2

### PR DESCRIPTION
**Pull request by bhedana**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 Gemfile.lock
+
+spec/fixtures/
```
Comment (edgester) [inline .gitignore]: Please fix lack of newline.

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 language: ruby
 rvm:
-  - 1.8.7
```
Comment (edgester) [inline .travis.yml]: Don't remove 1.8.7. It's the ruby version for Centos 6

File: .travis.yml [hunk 3]
```diff
@@ -10,8 +9,8 @@ script:
   - "bundle exec rake lint"
   - "bundle exec rake spec"
 env:
-  - PUPPET_VERSION=3.7.4
-  - PUPPET_VERSION=2.7.17
```
Comment (edgester) [inline .travis.yml]: Don't remove Puppet 2.7 and 3.7. 2.7 is the EPEL version of puppet for RHEL6

**Comment by bhedana** [comment]
restored travis yml

**Comment by edgester** [comment]
Please rebase and fix any conflicts.

**Comment by edgester** [comment]
I cherry-picked the following commits and resolved any conflicts:
702b852f8bac8d0044c1e01cab378555d0aa67b4
4e7b9ca51ee9501f958c2b2e4386a81b829ef3ee

The other three you'll need to rebase and resolve the conflicts.

**Pull request closed without merging**